tests/lean/run/coinductive_predicates.lean

```diff
@@ -0,0 +1,147 @@
+-- Coinductive predicate definition
```

Comment:
We need tests (in a separate test file) for all the ways users can screw up:

* Using `greatest_fixpoint`/`least_fixpoint` on non-preds.
* Using `greatest_fixpoint` on non-recursive definitions.
* Mutual recursion that wildy mixes the annotations. (The code that checks the annotation treats the first one as definite, I believe, so include tests both ways).